packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -379,44 +348,35 @@ class VisualStudio {
       }
     }
 
-    Map<String, dynamic>? usableVisualStudioDetails;
+    VswhereDetails? usableVisualStudioDetails;
     if (visualStudioDetails != null) {
       if (installationHasIssues(visualStudioDetails)) {
-        _cachedAnyVisualStudioDetails = visualStudioDetails;
+        _anyVisualStudioDetails = visualStudioDetails;
       } else {
         usableVisualStudioDetails = visualStudioDetails;
       }
     }
-    return usableVisualStudioDetails ?? <String, dynamic>{};
+    return usableVisualStudioDetails;
   }();
 
-  /// Returns the details dictionary of the latest version of Visual Studio,
-  /// regardless of components and version, or {} if no such installation is
+  /// Returns the details of the latest version of Visual Studio,
+  /// regardless of components and version, or null if no such installation is
   /// found.
   ///
-  /// If no installation is found, the cached VS details are set to an empty map
-  /// to avoid repeating vswhere queries that have already not found an
-  /// installation.
-  Map<String, dynamic>? _cachedAnyVisualStudioDetails;
-  Map<String, dynamic> get _anyVisualStudioDetails {
+  /// This may be overriden by `_usableVisualStudioDetails`'s initializer
+  /// to avoid repeating vswhere queries.
```

Comment:
Good question. The behavior between "best", "usable", and "any" is quite subtle especially since "usable"'s initializer may clobber "any" even if "any" was already initialized. Let me play around with this and get back to you

Reply:
@cbracken Done! I updated the `isUsable` check to include whether we validated the installation had satisfied Flutter's requirements (minimum supported version + required workloads). With that, we are able to remove `_usable` and `_any` details.